File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max=0
        container=[]
        for x in s:
            while x in container:
                container.pop(0)
            container.append(x)
            if len(container)>max:
                max = len(container)
        return max

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        ArrayList<Character> list=new ArrayList<>();
        int ans=0;

        for (int i=0; i<s.length(); i++) {
            while (list.contains(s.charAt(i))) {
                list.remove(0);
            }
            list.add(s.charAt(i));
            ans=Math.max(ans, list.size());
        }

        return ans;
    }
}